site/app/controllers/PollController.php

```diff
@@ -163,10 +163,25 @@ public function addNewPoll() {
         $poll_id = $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
         $file_path = null;
         if (isset($_FILES['image_file']) && $_FILES["image_file"]["name"] !== "") {
-            $file = $_FILES["image_file"];
-            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
-            move_uploaded_file($file["tmp_name"], $file_path);
-            $this->core->getQueries()->setPollImage($poll_id, $file_path);
+            $status = FileUtils::validateUploadedFiles($_FILES["image_file"]);
+            // validate the uploaded file size
+            if (array_key_exists("failed", $status)) {
+                unlink($file_path);
+                $this->core->getOutput()->renderResultMessage("Failed to validate uploads " . $status["failed"], false);
+            }
+            else {
+                $file = $_FILES["image_file"];
+                $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
+                move_uploaded_file($file["tmp_name"], $file_path);
+                // validate the uploaded file type is indeed an image
+                if (!FileUtils::isValidImage($file_path)) {
```

Comment:
I think checking that the file is a valid image should be done before copying it to the polls area, that way we can avoid the operation if its not an image file